extensions/aerosol_optics/mo_aerosol_optics.F90

```diff
@@ -0,0 +1,633 @@
+
+    !$acc data copyin(aero_type, aero_size, aero_mass, relhum)                         &
+    !$acc      create(atau, ataussa, ataussag) &
+    !$acc      create(aeromsk)
+    !$omp target data map(to:aero_type, aero_size, aero_mass, relhum) &
+    !$omp map(alloc:atau, ataussa, ataussag) &
+    !$omp map(alloc:aeromsk)
+    !
+    ! Aerosol mask; don't need aerosol optics if there's no aerosol
+    !
+    !$acc parallel loop gang vector default(none) collapse(2)
+    !$omp target teams distribute parallel do simd collapse(2)
+    do ilay = 1, nlay
+      do icol = 1, ncol
+        aeromsk(icol,ilay) = aero_type(icol,ilay) > 0
+      end do
+    end do
+
+    !
+    ! Aerosol size, relative humidity
+    !
+    if(check_values) then
+      if(any_vals_outside(aero_size, aeromsk, this%aero_bin_lo(1), this%aero_bin_hi(nbin))) &
+        error_msg = 'aerosol optics: requested aerosol size is out of bounds'
+      if(any_vals_outside(relhum, 0._wp, 1._wp)) &
+        error_msg = 'aerosol optics: relative humidity fraction is out of bounds'
+    end if
+    if(error_msg == "") then
+      !
+      !
+      ! ----------------------------------------
+      !
+      ! The lookup tables determining extinction coefficient, single-scattering albedo,
+      !   and asymmetry parameter g as a function of aerosol type, aerosol size and
+      !   relative humidity.
+      ! We compute the optical depth tau (= exintinction coeff * aerosol mass ) and the
+      !    products tau*ssa and tau*ssa*g separately for each aerosol type requested.
+      ! These are used to determine the aerosol optical properties.
+      !
+      if (allocated(this%aero_dust_ext)) then
+        !
+        ! Aerosol
+        !
+        call compute_all_from_table(ncol, nlay, nbnd, nbin, nrh,                                &
+                                    aeromsk, aero_type, aero_size, aero_mass, relhum,           &
+                                    this%aero_bin_lo, this%aero_bin_hi, this%aero_rh,           &
+                                    this%rh_nsteps, this%rh_step_sizes, this%rh_offsets,        &
+                                    this%aero_dust_ext, this%aero_dust_ssa, this%aero_dust_asy, &
+                                    this%aero_salt_ext, this%aero_salt_ssa, this%aero_salt_asy, &
+                                    this%aero_sulf_ext, this%aero_sulf_ssa, this%aero_sulf_asy, &
+                                    this%aero_bcar_rh_ext, this%aero_bcar_rh_ssa, this%aero_bcar_rh_asy, &
+                                    this%aero_bcar_ext, this%aero_bcar_ssa, this%aero_bcar_asy, &
+                                    this%aero_ocar_rh_ext, this%aero_ocar_rh_ssa, this%aero_ocar_rh_asy, &
+                                    this%aero_ocar_ext, this%aero_ocar_ssa, this%aero_ocar_asy, &
+                                    atau, ataussa, ataussag)
+
+      endif
+
+      !
+      ! Combine liquid and ice contributions into total cloud optical properties
```

Comment:
@mjiacono Can you revise the comment to make it appropriate for the aerosols?

Reply:
@RobertPiincus Comment changed.